### PR TITLE
feat: schedule compaction when adding sst files by editing region

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

as title

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/mito2/src/worker/handle_manifest.rs [hunk 1]
```diff
@@ -165,6 +169,26 @@ impl<S> RegionWorkerLoop<S> {
                 self.handle_region_edit(request).await;
             }
         }
+
+        if need_compaction {
```
Comment (killme2008) [inline src/mito2/src/worker/handle_manifest.rs:173]: I think it should be a request option, it's not a common practice to schedule compaction every time when editing regions.

Comment (MichaelScofield) [reply src/mito2/src/worker/handle_manifest.rs:173]: After discussed with @evenyag , we decide to add a config value to determine the time interval of two successive compactions. To align with the old behavior, the default value of this config is 0: no restrictions on how compactions are scheduled. In this editing region scene, people can adjust this config accordingly, to not trigger too frequent compactions.

File: src/mito2/src/engine/edit_region_test.rs [hunk 2]
```diff
@@ -15,20 +15,102 @@
...
+        compaction_time_window: None,
+        flushed_entry_id: None,
+        flushed_sequence: None,
+    };
+    engine
+        .edit_region(region.region_id, new_edit())
+        .await
+        .unwrap();
+    // Makes some time for the worker loop to process the request.
+    tokio::time::sleep(Duration::from_secs(1)).await;
```
Comment (evenyag) [inline src/mito2/src/engine/edit_region_test.rs:93]: Do we need this sleep?

File: src/mito2/src/region/opener.rs [hunk 3]
```diff
@@ -66,13 +66,15 @@ pub(crate) struct RegionOpener {
...
-    time_provider: Option<TimeProviderRef>,
+    time_provider: TimeProviderRef,
     stats: ManifestStats,
     wal_entry_reader: Option<Box<dyn WalEntryReader>>,
 }
 
 impl RegionOpener {
     /// Returns a new opener.
+    // TODO(LFC): Reduce the number of arguments.
+    #[allow(clippy::too_many_arguments)]
```
Comment (evenyag) [inline src/mito2/src/region/opener.rs:77]: Maybe we can leave it as a good first issue.

File: src/mito2/src/worker/handle_manifest.rs [hunk 4]
```diff
@@ -165,6 +169,36 @@ impl<S> RegionWorkerLoop<S> {
...
+            self.schedule_compaction(&region).await;
+        }
+    }
+
+    /// Schedule compaction for the region if necessary.
+    pub(crate) async fn schedule_compaction(&mut self, region: &MitoRegionRef) {
```
Comment (evenyag) [inline src/mito2/src/worker/handle_manifest.rs:179]: Maybe we should put this method in `handle_compaction.rs`.

File: src/mito2/src/engine/edit_region_test.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/mito2/src/engine/edit_region_test.rs:93]: Indeed not needed, removed.

File: src/mito2/src/region/opener.rs (hunk 3 above)
Comment (MichaelScofield) [reply src/mito2/src/region/opener.rs:77]: It requires some degree of understanding of the mito engine codes (to do the refactoring), I'm afraid it's not suitable for a "first" issue.

**Review by killme2008 — APPROVED** [review]

**Pull request merged**